### PR TITLE
Add additional boolean operations

### PR DESCRIPTION
**Pull request by devviktoria**
This pull requests adds the Crop and Trim boolean operations to Grahpite. 

The Trim operation is best described in the following video:
Adobe Illustrator Pathfinder Panel Explained! Understand Your Shape-Merging Options
https://youtu.be/Me5oOw8r0Is?t=370

The Crop operation is best described in this part of the video:
Adobe Illustrator Pathfinder Panel Explained! Understand Your Shape-Merging Options
https://youtu.be/Me5oOw8r0Is?t=428

Part of https://github.com/GraphiteEditor/Graphite/issues/4103

### CODE REVIEW

File: node-graph/nodes/path-bool/src/lib.rs [hunk 1]
```diff
@@ -36,24 +34,29 @@ async fn boolean_operation<I: graphic_types::IntoGraphicTable + 'n + Send + Clon
 
 	// The first index is the bottom of the stack
 	let flattened = flatten_vector(&content);
-	let mut result_vector_table = boolean_operation_on_vector_table(&flattened, operation);
+	let mut result_vector_table = match operation {
```
Comment (devviktoria) [inline node-graph/nodes/path-bool/src/lib.rs]: This code is created with AI help. For the existing Union, SubtractFront, SubstrackBack, Intersect and Difference we should use the boolean_operation_on_vector_table, but for Trim and Crop now we should use the cascading_subtract function.

**Comment by devviktoria** [comment]
So far I had tested only the basic functionality so I would like to test it with edge cases to make sure everything is working fine.  And also I will check Geminis code review and creating tests for the Crop and Trim.

**Comment by Keavon** [comment]
Happy to see this coming along, nice work. Beware of the merge conflict, you'll want to sync and resolve that when you get a chance. When you are ready to mark this as no longer a draft, you'll also get a round of reviews from the Cubic bot which will be helpful (and its reviews tend to have fewer false positives than Gemini which is more hit-or-miss).

**Comment by devviktoria** [comment]
Please note that I renamed BooleanOperation::Difference to BooleanOperation::Exclude. I did renamed the icon name in icons.ts however I could not rename the boolean-difference.svg because I noticed that is in a different repository. Please let me know if I should rename it in that other repository.

**Comment by devviktoria** [comment]
Basically I finished coding the Trim and the Crop boolean operations with some tests included too.

However I cannot finalize this without some feedback from the maintainers.

I have a question about the node's UI. Currently only the operation can be selected and the operation keeps the fill and the stroke of the shapes. The question is shall we give the user the option which appearance shall we keep for the end result, fill or the stroke or both? Or shall we keep this simple UI we currently have. Once I got the answer to this question I can finish the test code to test if the look is what it is expected.
extrabool_op_idea


There is a boolean-difference.svg in branding/assets/icon-16px-solid/. Since I had renamed the difference operation to Exclude (because the difference name was too confusing, for this operation the Exclude/Exclusion is the typical name in other applications.) This icon should be renamed to boolean-exclude.svg. But this icon is in another repository so I cannot rename it. Or this renaming is going to occur after this has been merged?

**Comment by devviktoria** [comment]
And also there are 2 new icons needed for this 2 boolean operation. (I forgot to mention this in my previous comment.)
